### ISSUE TITLE
simple API discussion about the AutoAugment

### ISSUE THREAD

**Issue by ain-soph** [pytorch/vision#3509]
## ❓ Questions and Help

question about the user interface API  
[transforms/autoaugment.py](https://github.com/pytorch/vision/blob/7b9d30eb7c4d92490d9ac038a140398e0a690db6/torchvision/transforms/autoaugment.py)  
The current usage would be `AutoAugment(AutoAugmentPolicy('cifar10'))`, but since the policy is just an `Enum`, I doubt whether it'll be more convenient to be `AutoAugment('cifar10')`. Is there any future advantage to use the policy?

cc @vfdev-5

**Comment by datumbox** [comment]
@ain-soph Thanks for the question.

The recommended way to use it is `AutoAugment(AutoAugmentPolicy.CIFAR10)` or `AutoAugment(AAP.CIFAR10)` if you shorten it during import (aka `from torchvision.transforms.autoaugment import AutoAugmentPolicy as AAP`).

But to address your main question of why we used an Enum instead of string, in the last release we made the decision to avoid using strings in favour of Enums. This happened in other places, for example with the introduction of `InterpolationModes`. There are a few reasons of why we opted for this:
- As we add typing in the library, using more enums makes things easier to document.
- Enums allow for easier find/replace in the code than strings and are less prone to typos.
- Enums give better integration with IDEs and allow for static analysis checks.

I'll close the issue to keep things clean but if you feel like the question has not been addressed feel free to reopen it and continue the discussion!

**Comment by ain-soph** [comment]
Thanks for your quick response. Another small typo fix is about the `AutoAugment` introduction in the 1.8.0 release blog:
https://pytorch.org/blog/pytorch-1.8-new-library-releases/
![image](https://user-images.githubusercontent.com/13214530/110225955-49130e80-7eb8-11eb-98a1-7a1ce7ec7802.png)

where `from torchvision import transforms` should be replaced by `import torchvision.transforms as transforms`
